File: Domain/Dto/Discord/DiscordGuildDto.cs
namespace Domain.Dto.Discord;

public record DiscordGuildDto(
    string GuildId,
    string GuildCreatorId,
    string GuildName,
    string? GuildIconUrl,
    List<DiscordUserDto> Members,
    List<DiscordChannelDto> VoiceChannels,
    List<DiscordChannelDto> TextChannels);

File: Frontend/src/model/discord/discordGuildDto.ts
import type { DiscordChannelDto } from "./discordChannelDto";
import type { DiscordUserDto } from "./discordUserDto";

export type DiscordGuildDto = {
    guildId: string;
    guildCreatorId: string;
    guildName: string;
    guildIconUrl?: string;
    members: DiscordUserDto[];
    voiceChannels: DiscordChannelDto[];
    textChannels: DiscordChannelDto[];
}